Buildings/Fluid/HeatExchangers/WetCoilEffectivenessNTU.mo

```diff
@@ -0,0 +1,583 @@
+     h=Medium2.h_default,
+     p=Medium2.p_default,
+     X=Medium2.X_default[1:Medium2.nXi]) "Default state for medium 2";
+
+initial equation
+  cp1_nominal = Medium1.specificHeatCapacityCp(sta1_default);
+  cp2_nominal = Medium2.specificHeatCapacityCp(sta2_default);
+  C1_flow_nominal = m1_flow_nominal*cp1_nominal;
+  C2_flow_nominal = m2_flow_nominal*cp2_nominal;
+  if (configuration == con.CrossFlowStream1MixedStream2Unmixed) then
+    flowRegime_nominal = if (C1_flow_nominal < C2_flow_nominal)
+      then
+        flo.CrossFlowCMinMixedCMaxUnmixed
+      else
+        flo.CrossFlowCMinUnmixedCMaxMixed;
+  elseif (configuration == con.CrossFlowStream1UnmixedStream2Mixed) then
+    flowRegime_nominal = if (C1_flow_nominal < C2_flow_nominal)
+      then
+        flo.CrossFlowCMinUnmixedCMaxMixed
+      else
+        flo.CrossFlowCMinMixedCMaxUnmixed;
+  elseif (configuration == con.ParallelFlow) then
+    flowRegime_nominal = flo.ParallelFlow;
+  elseif (configuration == con.CounterFlow) then
+    flowRegime_nominal = flo.CounterFlow;
+  elseif (configuration == con.CrossFlowUnmixed) then
+    flowRegime_nominal = flo.CrossFlowUnmixed;
+  else
+    // Invalid flow regime. Assign a value to flowRegime_nominal, and stop with an assert
+    flowRegime_nominal = flo.CrossFlowUnmixed;
+    assert(configuration >= con.ParallelFlow and
+      configuration <= con.CrossFlowStream1UnmixedStream2Mixed,
+      "Invalid heat exchanger configuration.");
+  end if;
+
+equation
+  if allowFlowReversal1 then
+    fra_a1 = Modelica.Fluid.Utilities.regStep(
+      m1_flow,
+      1,
+      0,
+      m1_flow_small);
+    fra_b1 = 1-fra_a1;
+  else
+    fra_a1 = 1;
+    fra_b1 = 0;
+  end if;
+  if allowFlowReversal2 then
+    fra_a2 = Modelica.Fluid.Utilities.regStep(
+      m2_flow,
+      1,
+      0,
+      m2_flow_small);
+    fra_b2 = 1-fra_a2;
+  else
+    fra_a2 = 1;
+    fra_b2 = 0;
+  end if;
+  // Assign the flow regime for the given heat exchanger configuration and
+  // mass flow rates
+  if (configuration == con.ParallelFlow) then
+    flowRegime = if (C1_flow*C2_flow >= 0)
+      then
+        flo.ParallelFlow
+      else
+        flo.CounterFlow;
+  elseif (configuration == con.CounterFlow) then
+    flowRegime = if (C1_flow*C2_flow >= 0)
+      then
+        flo.CounterFlow
+      else
+        flo.ParallelFlow;
+  elseif (configuration == con.CrossFlowUnmixed) then
+    flowRegime = flo.CrossFlowUnmixed;
+  elseif (configuration == con.CrossFlowStream1MixedStream2Unmixed) then
+    flowRegime = if (C1_flow < C2_flow)
+      then
+        flo.CrossFlowCMinMixedCMaxUnmixed
+      else
+        flo.CrossFlowCMinUnmixedCMaxMixed;
+  else
+    // have ( configuration == con.CrossFlowStream1UnmixedStream2Mixed)
+    flowRegime = if (C1_flow < C2_flow)
+      then
+        flo.CrossFlowCMinUnmixedCMaxMixed
+      else
+        flo.CrossFlowCMinMixedCMaxUnmixed;
+  end if;
+
```

Comment:
Most of this code is a duplicate of the code from `Fluid.HeatExchangers.BaseClasses.PartialEffectivenessNTU` and `Fluid.HeatExchangers.BaseClasses.PartialEffectiveness`.
See my remark above regarding the refactoring that I think is necessary here.

Reply:
I agree. Let's talk to Michael Wetter.